### PR TITLE
Gets the remove icon from theme

### LINKED ISSUE #6870: Allow theme to specify Tag remove icon
### Expected Behavior

Theme should support `tag.icons.remove`. (mirrors select / menu structure)

### Actual Behavior

FormClose is built into Grommet.

Follow-on work from: https://github.com/grommet/hpe-design-system/issues/3467

### PR DESCRIPTION
**Pull request by vinodkv2511**
#### What does this PR do?
Allows the remove icon in `Tag` component to be provided from the theme. Fixes #6870 

#### Where should the reviewer start?
Start with the `Tag.js` file

#### What testing has been done on this PR?
1. Ran unit tests and failures
2. Updated snapshot
3. Manually tested in storybook by updating the icon from theme and is working fine.

#### How should this be manually tested?
1. Run storybook and navigate to `Tag > onRemove` story
2. Verify the `FormClose` icon is rendered by default
3. Update the icon(`tag.icons.remove`) from `themes > base.js` and see the changes reflected on the story
4. Repeat the same with `Tag > Size` story as well

#### Do Jest tests follow these best practices?
~~Not applicable as no new tests are added~~ Added a new test.

- [ ] `screen` is used for querying.
- [x] The correct query is used. (Refer to [this list of queries](https://testing-library.com/docs/queries/about/#priority)).
- [ ] `userEvent` is used in place of `fireEvent`.
- [ ] `asFragment()` is used for snapshot testing.

#### What are the relevant issues?
#6870 

#### Do the grommet docs need to be updated?
Yes. https://v2.grommet.io/tag#theme

#### Should this PR be mentioned in the release notes?
Yes.

#### Is this change backwards compatible or is it a breaking change?
This is backward compatible

### CODE REVIEW

File: src/js/themes/base.d.ts [hunk 1]
```diff
@@ -1617,6 +1617,9 @@ export interface ThemeType {
       };
     };
     value?: TextProps;
+    icons?: {
+      remove?: any;
```
Comment (jcfilben) [inline src/js/themes/base.d.ts:1621]: Can we adjust this type to be more specific
```suggestion
      remove?: React.ReactNode | Icon;
```

**Review by jcfilben — COMMENTED** [review]
Can you add a jest test that specifies a different icon for the Tag remove icon in the theme?

**Comment by vinodkv2511** [comment]
Hi @jcfilben - Is there anything else needed from me on this PR?

**Review by britt6612 — APPROVED** [review]

File: src/js/components/Tag/Tag.js [hunk 2]
```diff
@@ -14,6 +12,7 @@ import { StyledRemoveButton, StyledTagButton } from './StyledTag';
 const Tag = forwardRef(
   ({ name, value, size, onRemove, onClick, ...rest }, ref) => {
     const theme = useContext(ThemeContext) || defaultProps.theme;
+    const RemoveIcon = theme.tag.icons.remove;
```
Comment (MikeKingdom) [inline src/js/components/Tag/Tag.js:15]: Really this should be more tolerant and have a fallback. How about:
```jsx
const RemoveIcon = theme.tag?.icons?.remove || FormClose;
```

Comment (vinodkv2511) [reply src/js/components/Tag/Tag.js]: Added fallback and added a test.

**Comment by vinodkv2511** [comment]
Hi @jcfilben @MikeKingdom ,
Circle CI bundlesize is failing. I don't think it is related to this issue. 
Below is the screenshot of error.
image

File: src/js/components/Tag/Tag.js [hunk 3]
```diff
@@ -1,7 +1,6 @@
 import React, { forwardRef, useContext } from 'react';
 import { ThemeContext } from 'styled-components';
-
-import { FormClose } from 'grommet-icons/icons/FormClose';
+import { FormClose } from 'grommet-icons';
```
Comment (britt6612) [inline src/js/components/Tag/Tag.js:3]:
```suggestion
import { FormClose } from 'grommet-icons/icons/FormClose';
```

Comment (britt6612) [reply src/js/components/Tag/Tag.js:3]: @vinodkv2511 should fix the error..

Comment (vinodkv2511) [reply src/js/components/Tag/Tag.js]: Updated!

Comment (vinodkv2511) [reply src/js/components/Tag/Tag.js]: image


This time the bundle size is 0.1KB more than original. @britt6612

Comment (vinodkv2511) [reply src/js/components/Tag/Tag.js]: @jcfilben  @britt6612 any way to fix this? Or is there a way to increase the bundle size?

Comment (jcfilben) [reply src/js/components/Tag/Tag.js]: Hey @vinodkv2511 in the `package.json` file you can set the `bundlesize.maxSize` property to "162 kB"

Comment (vinodkv2511) [reply src/js/components/Tag/Tag.js]: Thanks @jcfilben. I have update it to 162KB.

**Review by jcfilben — APPROVED** [review]

File: src/js/components/Tag/Tag.js [hunk 4]
```diff
@@ -14,6 +13,7 @@ import { StyledRemoveButton, StyledTagButton } from './StyledTag';
 const Tag = forwardRef(
   ({ name, value, size, onRemove, onClick, ...rest }, ref) => {
     const theme = useContext(ThemeContext) || defaultProps.theme;
+    const RemoveIcon = theme.tag.icons.remove || FormClose;
```
Comment (MikeKingdom) [inline src/js/components/Tag/Tag.js:16]: Still needs this change:
```suggestion
    const RemoveIcon = theme.tag.icons?.remove || FormClose;
```

**Review by MikeKingdom — APPROVED** [review]

**Pull request merged**